go.mod

```diff
@@ -5,12 +5,14 @@ go 1.22.0
 toolchain go1.23.4
 
 require (
+	cloud.google.com/go/kms v1.20.2
```

Comment:
@bradegler Ooof - I'm worried about all these new dependencies. One of the advantages of using a `crypto.Signer` interface is that we can push this out into another package. This feels like a lot of dependencies to take/maintain across all of `pkg` customers just to avoid these LOC:

```go
	client, err := kms.NewKeyManagementClient(ctx)
	if err != nil {
		return nil, fmt.Errorf("failed to setup client: %w", err)
	}
	signer, err := gcpkms.NewSigner(ctx, client, keyID)
	if err != nil {
		return nil, fmt.Errorf("failed to create signer: %w", err)
	}
	return signer, nil
```

Reply:
That's fair. I just didn't feel like reimplementing this every time we used it but the extra dependencies do suck. I see you already fixed it.

Reply:
Yea - I never complain without accompanying code :). Also, I just updated all the deps on go-gcpkms too.